src/tests/ftest/util/command_utils.py

```diff
@@ -1048,6 +1062,180 @@ def get_config_value(self, name):
             value = self.manager.job.get_config_value(name)
         return value
 
+    def get_current_state(self):
+        """Get the current state of the daos_server ranks.
+
+        Returns:
+            dict: dictionary of server rank keys, each referencing a dictionary
+                of information containing at least the following information:
+                    {"host": <>, "uuid": <>, "state": <>}
+                This will be empty if there was error obtaining the dmg system
+                query output.
+
+        """
+        data = {}
+        ranks = {host: rank for rank, host in enumerate(self._hosts)}
+        if not self._verify_socket_dir:
+            command = "systemctl is-active {}".format(
+                self.manager.job.service_name)
+        else:
+            command = "prep {}".format(self.manager.job.command)
+        task = run_task(self._hosts, command, 30)
+        for output, nodelist in task.iter_buffers():
+            for node in nodelist:
+                data[ranks[node]] = {
+                    "host": node, "uuid": "-", "state": str(output)}
+        return data
+
+    def update_expected_states(self, ranks, state):
+        """Update the expected state of the specified job rank.
+
+        Args:
+            ranks (object): job ranks to update. Can be a single rank (int),
+                multiple ranks (list), or all the ranks (None).
+            state (object): new state to assign as the expected state of this
+                rank. Can be a str or a list of str.
+        """
+        if ranks is None:
+            ranks = [key for key in self._expected_states]
```

Comment:
(pylint-unnecessary-comprehension) Unnecessary use of a comprehension

Reply:
Fixed